### PR TITLE
docs: Release notes for 0.15.0.

### PR DESCRIPTION
**Pull request by justin-determined-ai**
## Commentary (optional)

Release notes for 0.14.6 were never merged back into master, so this PR includes both. Will be fixed post-release.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: docs/release-notes.txt [hunk 1]
```diff
@@ -4,10 +4,103 @@
...
+**************
+
+Version 0.15.0
+==============
+
+**Release Date:** April 14, 2021
+
+**New Features**
+
+-  WebUI: Provide historical utilization data on the Cluster page. This
```
Comment (vishnu2kmohan) [inline docs/release-notes.txt:18]:
```suggestion
-  WebUI: Provide historical allocation data on the Cluster page. This
```

**Review by vishnu2kmohan — APPROVED** [review]
Just a minor suggestion, and once we add a note about the PyTorch checkpoints no longer depending on `cloudpickle`, we should be good.

File: docs/release-notes.txt (hunk 1 above)
Comment (justin-determined-ai) [reply docs/release-notes.txt:18]: @mackrorysd is the cloudpickle change transparent to the user?

Comment (mackrorysd) [reply docs/release-notes.txt:18]: Yes it should be. Adding a relnote here: https://github.com/determined-ai/determined/pull/2227

Comment (justin-determined-ai) [reply docs/release-notes.txt:18]: I included #2227 in this PR!

**Review by vishnu2kmohan — APPROVED** [review]

**Pull request merged**